From 5322093d44e20fa69cf4e34dca915e1f5bf2e639 Mon Sep 17 00:00:00 2001
From: Nick Liu <nick_liu@avalue.com.tw>
Date: Thu, 26 May 2016 15:18:13 +0800
Subject: [PATCH 1/1] Fixed can bus

modified:   kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
modified:   kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
---
 .../arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts  |  4 ++++
 kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi        | 24 ++++++++++++++++++----
 2 files changed, 24 insertions(+), 4 deletions(-)

diff --git a/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts b/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
index 981bb6b..5a801eb 100644
--- a/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
+++ b/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
@@ -51,9 +51,13 @@
 };
 
 &can1 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_flexcan1>;
     status = "okay";
 };
 
 &can2 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_flexcan2>;
     status = "okay";
 };
diff --git a/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi b/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
index 3ef2713..f07f3f4 100644
--- a/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
+++ b/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
@@ -185,6 +185,22 @@
 		status = "okay";
 	};
 
+    v4l2_cap_0 {
+        compatible = "fsl,imx6q-v4l2-capture";
+        ipu_id = <0>;
+        csi_id = <0>;
+        mclk_source = <0>;
+        status = "okay";
+    };
+
+    v4l2_cap_1 {
+        compatible = "fsl,imx6q-v4l2-capture";
+        ipu_id = <0>;
+        csi_id = <1>;
+        mclk_source = <0>;
+        status = "okay";
+    };
+
 	v4l2_out {
 		compatible = "fsl,mxc_v4l2_output";
 		status = "okay";
@@ -430,15 +446,15 @@
 	imx6qdl-smarc {
 		pinctrl_flexcan1: flexcan1grp {
 			fsl,pins = <
-				MX6QDL_PAD_GPIO_7__FLEXCAN1_TX		0x80000000
-				MX6QDL_PAD_GPIO_8__FLEXCAN1_RX		0x80000000
+                MX6QDL_PAD_GPIO_7__FLEXCAN1_TX		0x80000000
+                MX6QDL_PAD_GPIO_8__FLEXCAN1_RX		0x80000000
 			>;
 		};
 
 		pinctrl_flexcan2: flexcan2grp {
 			fsl,pins = <
-				MX6QDL_PAD_KEY_COL4__FLEXCAN2_TX	0x80000000
-				MX6QDL_PAD_KEY_ROW4__FLEXCAN2_RX	0x80000000
+                MX6QDL_PAD_KEY_COL4__FLEXCAN2_TX	0x80000000
+                MX6QDL_PAD_KEY_ROW4__FLEXCAN2_RX	0x80000000
 			>;
 		};
 
-- 
1.9.1

